magma/digital.py

```diff
@@ -206,19 +206,20 @@ def undriven(self):
         m.wire(DefineUndriven()().O, self)
 
 
-def make_Define(name, port, direction):
-    def simulate(self, value_store, state_store):
-        pass
-
+def make_Define(_name, port, direction):
     @lru_cache(maxsize=None)
     def DefineCorebit():
-        return m.circuit.DeclareCoreirCircuit(
-            f"corebit_{name}",
-            port, direction(Digital),
-            coreir_name=name,
-            coreir_lib="corebit",
-            simulate=simulate
-        )
+        class _Primitive(m.Circuit):
+            renamed_ports = m.circuit.coreir_port_mapping
+            name = f"corebit_{_name}"
+            coreir_name = _name
+            coreir_lib = "corebit"
+
+            def simulate(self, value_store, state_store):
+                pass
+
+            io = m.IO(**{port: direction(Digital)})
+        return _Primitive
```

Comment:
Refactor to avoid deprecation warning from old IO syntax (which is used implicitly by declare circuit.

Reply:
Great!